pages/article_list_page.rb

```diff
@@ -4,11 +4,15 @@ class ArticleListPage < DemoAppPage
   validate :title, /\ADemo web application - Listing Articles\z/
 
   element :new_article_button, :xpath, "//a[@href='/articles/new']"
-  element :article_button, :xpath, ->(title) { "//a[contains(.,'#{title}')]" }
-  element :destroy_button, :xpath,
-          ->(title) { "//strong[.='#{title}']/following-sibling::a[normalize-space(.)='Destroy'][1]" }
-  element :edit_button, :xpath,
-          ->(title) { "//strong[.='#{title}']/following-sibling::a[normalize-space(.)='Edit'][1]" }
+  element :article_button, :xpath, ->(title) { "(//a[contains(.,'#{title}')])[1]" }
+  element :destroy_button, :xpath, lambda { |title|
```

Comment:
Problem with selectors. They are complex and not flexible

Reply:
After investigation it was found not structured HTML. Please ask frontend developer to nest action buttons into article_item section